### ISSUE TITLE
$sanitizer->selectorValue changes the effect of values in advanced text search #=

### ISSUE THREAD

**Issue by 16th-earl** [processwire/processwire-issues#1395]
### Short description of the issue

$sanitizer->selectorValue changes the values used in advanced text search in a way that can cause this search to not work as expected. It strips out leading plus signs and removes double-quotes in some circumstances.

### Expected behavior

$sanitizer->selectorValue should preserve the selector value so it still performs as expected when using the advanced text search operator (#=).

### Actual behavior

In some cases values are changed in a way that alters the effect of an advanced text search.

```
// Discards the leading plus sign, search doesn't work as expected.
$sanitizer->selectorValue('+foo bar') === 'foo bar';

// Keeps double-quotes; phrase searching then doesn't work as expected.
$sanitizer->selectorValue('"foo bar baz"') === '"foo bar baz"';

// Discards double-quotes; phrase searching then doesn't work as expected.
$sanitizer->selectorValue('foo "bar baz"') === 'foo bar baz'; 
```
### Optional: Suggestion for a possible fix

Double quotes could be translated to parentheses. This code works, though there may be something more efficient.

```
$paren = ')';
for ($i = strlen($value) - 1; $i >= 0; $i--) {
    if ($value[$i] === '"') {
        $value[$i] = $paren;
        $paren = $paren === ')' ? '(' : ')';
    }
}
```

### Setup/Environment

* ProcessWire version: 3.0.164
* PHP version: 7.2

**Comment by teppokoivula** [comment]
Just for reference, similar issue was discussed earlier in https://github.com/processwire/processwire-issues/issues/1234.

**Comment by 16th-earl** [comment]
Thanks! That’s an interesting discussion. I didn’t find it earlier because I only searched through open issues before logging this report.

If this is going to be closed as an issue (like #1234 was) then I think there should be a documentation change to the [selectorValue](https://processwire.com/api/ref/sanitizer/selector-value/) method and [selectors](https://processwire.com/docs/selectors/operators/#contains-advanced) document so that people know to not just routinely use this method on a selector value used with the advanced text search operator.

Ideally, I think we should be able to just use $sanitizer->selectorValue() without having to think about the type of text search that’s being done. Because the $sanitizer functions are so important for keeping the system secure I think they should be particularly easy to use.

**Comment by ryancramerdesign** [comment]
@16th-earl @teppokoivula 

> Ideally, I think we should be able to just use $sanitizer->selectorValue() without having to think about the type of text search that’s being done. 

Yeah I think that would be ideal too. Though the `selectorValue()` method isn't going to know whether or not you are using an advanced search operator unless you tell it. Since the rules are different for normal selector values than they are for advanced text search operator selector values, maybe the best bet is just to have another method for it. I've gone ahead and attempted to make one: `$sanitizer->selectorValueAdvanced()`. I'm not sure it's perfect yet but it does at least seem to work correctly with all the examples you mentioned earlier, as well as all those I could think of in testing here.
